### ISSUE TITLE
How to get top-5 or top-3 accuracy by ConfusionMatrix?

### ISSUE THREAD

**Issue by arashno** [torch/optim#140]
Hello all,
Can I use ConfusionMatrix to compute top-n accuracy for my classifier or not?
If yes, how can I do that? Is there any sample code to do that?
Thanks

**Comment by taineleau** [comment]
hi,

I think top-k acc. need to hack the `ConfusionMatrix.lua` file.

And there're two ways to calculate top-k acc., hope the following helps.

1. please kindly check the [resnet](https://github.com/facebook/fb.resnet.torch/blob/master/train.lua#L107) training script L107. 

2. Torchnet's utils for [topk](https://github.com/torchnet/torchnet#tntclasserrormeterself-topk-accuracy).
